ecosystem/typescript/sdk/examples/typescript/gas_fee_payer.ts

```diff
@@ -0,0 +1,153 @@
+  console.log("\n=== Alice and Bob accounts are randomly generated ===");
+  console.log("Alice's profile:");
+  console.log(`  alice:`);
+  console.log(`    private_key: "${alice.toPrivateKeyObject().privateKeyHex}"`);
+  console.log(`    public_key: "${alice.pubKey()}"`);
+  console.log(`    account: ${alice.address()}`);
+  console.log(`    rest_url: "https://fullnode.devnet.aptoslabs.com"`);
+  console.log(`    faucet_url: "https://faucet.devnet.aptoslabs.com"`);
+
+  console.log("Bob's profile:");
+  console.log(`  bob:`);
+  console.log(`    private_key: "${bob.toPrivateKeyObject().privateKeyHex}"`);
+  console.log(`    public_key: "${bob.pubKey()}"`);
+  console.log(`    account: ${bob.address()}`);
+  console.log(`    rest_url: "https://fullnode.devnet.aptoslabs.com"`);
+  console.log(`    faucet_url: "https://faucet.devnet.aptoslabs.com"`);
+  await waitForEnter();
+
+
+  // Fund Alice and Bob accounts.
+  console.log("\n=== Alice and Bob accounts are funded ===");
+  await faucetClient.fundAccount(alice.address(), 100_000_000);
+  await faucetClient.fundAccount(bob.address(), 100_000_000);
+  console.log(`Alice's balance: ${await getBalance(alice)} octas`)
+  console.log(`Bob's balance: ${await getBalance(bob)} octas`)
+  await waitForEnter();
+
+
+  // Create collection and token on Alice's account
+  console.log("\n=== Alice created a collection and a token ===");
+  const collectionName = "AliceCollection";
+  const tokenName = "Alice Token";
+  await ensureTxnSuccess(
+      tokenClient.createCollection(
+          alice,
+          collectionName,
+          "Alice's simple collection",
+          "https://aptos.dev"
+      )
+  );
+  await ensureTxnSuccess(
+    tokenClient.createToken(
+      alice,
+      collectionName,
+      tokenName,
+      "Alice's simple token",
+      1,
+      "https://aptos.dev/img/nyan.jpeg",
+      1000,
+      alice.address(),
+      0,
+      0,
+      ["key"],
+      ["2"],
+      ["u64"],
+    ),
+  );
+  const propertyVersion = 0;
+  const tokenId = {
+    token_data_id: {
+      creator: alice.address().hex(),
+      collection: collectionName,
+      name: tokenName,
+    },
+    property_version: `${propertyVersion}`,
+  };
+  await tokenClient.getCollectionData(alice.address().hex(), collectionName);
+  let aliceToken = await tokenClient.getTokenForAccount(alice.address().hex(), tokenId);
+  console.log(`Alice's token amount: ${aliceToken.amount}`)
+  assert(aliceToken.amount == "1");
+  console.log(`Alice's balance: ${await getBalance(alice)} octas`)
+  console.log(`Bob's balance: ${await getBalance(bob)} octas`)
+  await waitForEnter();
+
+
+  // Transfer Token from Alice's Account to Bob's Account with bob paying the fee
+  console.log("\n=== Alice sent the token to Bob while Bob paied the fee ===");
+  const aliceBefore = await getBalance(alice);
+  const bobBefore = await getBalance(bob);
+  const txnHash = await tokenClient.directTransferTokenWithFeePayer(
```

Comment:
It's a good start to use directTransferTokenWithFeePayer which supports a separate gas payer. But we should add another type of tx here where you have to construct a tx with a gas payer also to show people how to do it. directTransferTokenWithFeePayer is a function only used for token v1 and not a common one either. Most people using gas payer would use it for a different transaction

Reply:
yes directtransfer was my example on how to construct the tx. You should look in that to see how to make a general tx.

Reply:
Added another use of fee payer transaction (`createTokenWithFeePayer`). It builds, signs, and submits a raw transaction with a fee payer. One should be able to follow this flow for their own type of transactions.